pyfar/dsp/fft.py

```diff
@@ -105,7 +105,110 @@ def irfft(spec, n_samples, sampling_rate, fft_norm):
+
+    Parameters
+    ----------
+    n_samples : int
+        The number of samples in the signal
+    sampling_rate : int
+        The sampling rate of the signal
+
+    Returns
+    -------
+    frequencies : array, double
+        The positive discrete frequencies in Hz for which the FFT is
+        calculated.
+    """
+    return sfft.fftshift(sfft.fftfreq(n_samples, d=1/sampling_rate))
+
+
+def fft(data, n_samples, sampling_rate, fft_norm):
+    """
+    Calculate the FFT of a complex-valued time-signal.
+
+    The function returns the double sided spectrum. The normalization is
+    considered according to ``'fft_norm'`` as described in
+    :py:func:`~pyfar.dsp.fft.normalization`
+    and :py:mod:`FFT concepts <pyfar._concepts.fft>`.
+
+    Parameters
+    ----------
+    data : array, double
+        Array containing the time domain signal with dimensions
+        (..., ``'n_samples'``)
+    n_samples : int
+        The number of samples
+    sampling_rate : number
+        sampling rate in Hz
+    fft_norm : 'none', 'unitary', 'amplitude', 'rms', 'power', 'psd'
+        See documentation of :py:func:`~pyfar.dsp.fft.normalization`.
+
+    Returns
+    -------
+    spec : array, complex
+        The complex valued right-hand side of the spectrum with dimensions
+        (..., n_bins)
+
+    """
+
+    # DFT
+    spec = sfft.fft(
+        data, n=n_samples, axis=-1, workers=multiprocessing.cpu_count())
+    # Normalization
+    spec = normalization(spec, n_samples, sampling_rate, fft_norm,
+                         inverse=False, single_sided=False)
+
+    return spec
+
+
+def ifft(spec, n_samples, sampling_rate, fft_norm):
+    """
+    Calculate the IFFT of a double-sided Fourier spectrum.
+
+    The function takes double-sided spectrum and returns a  time signal.
+    The normalization is considered according to
+    ``'fft_norm'`` as described in :py:func:`~pyfar.dsp.fft.normalization`
+    and :py:mod:`FFT concepts <pyfar._concepts.fft>`.
+
+    Parameters
+    ----------
+    spec : array, complex
+        The complex valued right-hand side of the spectrum with dimensions
+        (..., n_bins)
+    n_samples : int
+        The number of samples of the corresponding time signal. This is crucial
+        to allow for the correct transform of time signals with an odd number
+        of samples.
+    sampling_rate : number
+        sampling rate in Hz
+    fft_norm : 'none', 'unitary', 'amplitude', 'rms', 'power', 'psd'
+        See :py:func:`~pyfar.dsp.fft.normalization`.
+
+    Returns
+    -------
+    data : array, double
+        Array containing the time domain signal with dimensions
+        (..., ``'n_samples'``)
+    """
+
+    # Inverse normalization
+    spec = normalization(spec, n_samples, sampling_rate, fft_norm,
+                         inverse=True, single_sided=False)
+    # Inverse DFT
+    data = sfft.ifft(
         spec, n=n_samples, axis=-1, workers=multiprocessing.cpu_count())
```

Comment:
Same comment as above, the shifted spectrum needs to be shifted back before computing the inverse transform.
```suggestion
    data = sfft.ifft(
        sfft.ifftshift(spec, axes=-1),
        n=n_samples, axis=-1, workers=multiprocessing.cpu_count())
```